guide_drafts/model-derives.md

```diff
@@ -0,0 +1,926 @@
+  )).execute(&db_connection);
+  
+  let updated_hero = users.first(&db_connection)::<Users>
+      .expect("Error loading first user");
+  
+  println!("Our Hero's update name: {} {}", updated_hero.first_name, updated_hero.last_name); // > Our Hero's updated name: Alyx Vance
+}
+```
+
+If we were to try and call `id()` and also update our record without implementing `Identifiable`, we would get the following errors.
+
+```rust
+error[E0599]: no method named `id` found for type `diesel_demo_cli::models::User` in the current scope
+  --> src/bin/main.rs:37:48
+   |
+37 |     println!("Our Hero's Id: {}", hero.id());
+   |                                        ^^ field, not a method
+   |
+   = help: did you mean to write `hero.id` instead of `hero.id(...)`?
+  --> src/bin/main.rs:39:5
+   |
+39 |     diesel::update(&hero).set((
+   |     ^^^^^^^^^^^^^^ the trait `diesel::associations::HasTable` is not implemented for `diesel_demo_cli::models::User`
+   |
+   = note: required because of the requirements on the impl of `diesel::associations::HasTable` for `&diesel_demo_cli::models::User`
+   = note: required because of the requirements on the impl of `diesel::query_builder::IntoUpdateTarget` for `&diesel_demo_cli::models::User`
+   = note: required by `diesel::update`
+
+error[E0277]: the trait bound `&diesel_demo_cli::models::User: diesel::Identifiable` is not satisfied
+  --> src/bin/main.rs:39:5
+   |
+39 |     diesel::update(&hero).set((
+   |     ^^^^^^^^^^^^^^ the trait `diesel::Identifiable` is not implemented for `&diesel_demo_cli::models::User`
+   |
+   = note: required because of the requirements on the impl of `diesel::query_builder::IntoUpdateTarget` for `&diesel_demo_cli::models::User`
+   = note: required by `diesel::update`
+```
+
+From this error,
+you can see the compiler assumes we are looking for a field on the struct,
+but what we want is just the value of the primary key,
+which happens to also be the `id` field.
+The compiler is also giving us some hints about trait bounds in each one of those *note* sections.
+The first two notes about `HasTable` and `IntoUpdateTarget` might be helpful if we refer to the [Diesel API docs](http://docs.diesel.rs/diesel/associations/trait.Identifiable.html),
+but the third error lets us know exactly which trait isn't satisfied! :smile:
+
+The following code is generated by `cargo expand` to get our example code working again.
+
+```rust
+#[derive(Queryable)]
+pub struct User {
+    pub id: i32,
+    pub first_name: String,
+    pub last_name: String,
+    pub email: String,
+}
+
+impl <'ident> Identifiable for &'ident User {
+    type Id = (&'ident i32);
+
+    fn id(self) -> Self::Id {
+        (&self.id)
+    }
+}
+
+impl HasTable for User {
+    type Table = users::table;
+
+    fn table() -> Self::Table {
+        users::table
+    }
+}
+```
+
+## <a name="aschangeset"></a>AsChangeset
+
+For more slightly more complicated database updates,
+`AsChangeset` can make your development easier.
```

Comment:
> make your development easier

it that something people say? Would've gone with "make your code more ergonomic"